src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -60,14 +60,14 @@ internal unsafe void AppendExtraBuffer(byte* buffer, int bufferLength)
             {
                 Debug.Assert(_endIndex > 0);
                 int spaceRemaining = _unprocessedBufferToBeRead.Length - _endIndex;
-                if (spaceRemaining < charLen)
+                if (spaceRemaining < chars.Length)
```

Comment:
I didn't change it so as not to mix purposes in this PR, but the logic in this method is suspect, in particular around ensuring there's enough space in the buffer.